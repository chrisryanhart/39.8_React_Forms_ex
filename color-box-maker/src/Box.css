.Box {
    /* height: 100px;
    width: 100px;
    background-color: greenyellow; */
    border: solid black 2px;
}