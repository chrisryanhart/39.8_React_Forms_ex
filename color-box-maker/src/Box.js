import React from 'react';
import './Box.css';

const Box = ({id, remove, width="100px", height='100px', backgroundColor='greenyellow'}) => {
    // have to remove the box upon click

    const handleClick = () => remove(id);
    
    return (
        <>
           <div className='Box' key={id} style={{width, height, backgroundColor}}/>
           <button onClick={handleClick}>X</button>
        </>
    );
}

export default Box;