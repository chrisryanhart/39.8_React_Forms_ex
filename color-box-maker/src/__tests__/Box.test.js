import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Box from '../Box';

it("renders without crashing", function() {
    render(<Box/>);
})

it("renders without crashing", function() {
    const {asFragment} = render(<Box/>);
    expect(asFragment()).toMatchSnapshot();
})
