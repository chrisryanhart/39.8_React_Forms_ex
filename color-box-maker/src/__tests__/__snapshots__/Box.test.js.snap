// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 100px; height: 100px; background-color: greenyellow;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
